sqlite/lib/SQLiteService.js

```diff
@@ -15,6 +15,19 @@ class SQLiteService extends SQLService {
         dbc.function('session_context', key => dbc[$session][key])
         dbc.function('regexp', { deterministic: true }, (re, x) => (RegExp(re).test(x) ? 1 : 0))
         dbc.function('ISO', { deterministic: true }, d => d && new Date(d).toISOString())
+
+        // define date and time functions in js to allow for throwing errors
+        const isTime = /^\d{1,2}:\d{1,2}:\d{1,2}$/
+        const hasTimezone = /([+-]\d{1,2}:?\d{0,2}|Z)$/
+        const throwNaN = n => { if (Number.isNaN(n)) { throw new Error(`Value does not contain a date "${n}"`) } else { return n } }
+        const toDate = d => new Date(isTime.test(d) ? `1970-01-01T${d}Z` : hasTimezone.test(d) ? d : d + 'Z')
+        dbc.function('year', { deterministic: true }, d => d && throwNaN(new Date(d).getUTCFullYear()))
+        dbc.function('month', { deterministic: true }, d => d && throwNaN(new Date(d).getUTCMonth()) + 1)
```

Comment:
What happens for invalid date values? Doesn't the new Date constructor reject that?

Reply:
`throwNaN` is what make Invalid Dates throw. Other wise it will just become `NaN`.